### PR TITLE
Exchanges: Bypass websocket book validation

### PR DESCRIPTION
**Pull request by shazbert**
# PR Description

Reduces the need to traverse and validate book contents when there is a checksum present from the exchange.

Exchanges affected:
- [x] Bitfinex
- [x] FTX
- [x] Kraken
- [x] Okex
- [x] OkCoin

## Type of change

Please delete options that are not relevant and add an `x` in `[]` as item is complete.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How has this been tested

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration and
also consider improving test coverage whilst working on a certain feature or package.

- [x] go test ./... -race
- [ ] golangci-lint run
- [ ] Test X

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation and regenerated documentation via the documentation tool
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally and on Travis with my changes
- [n/a] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by gloriousCode — COMMENTED** [review]
This seems like something that would be good to add as a config variable per exchange to allow someone to have the choice and not have to recompile the whole application.

File: exchanges/orderbook/orderbook_types.go [hunk 1]
```diff
@@ -79,10 +79,15 @@ type Base struct {
...
 	ExchangeName string        `json:"exchangeName"`
+
 	// NotAggregated defines whether an orderbook can contain duplicate prices
 	// in a payload
 	NotAggregated bool `json:"-"`
 	IsFundingRate bool `json:"fundingRate"`
+
+	// ChecksumBypass defines an allowance to bypass internal verification if
+	// the book has been verified by checksum.
+	ChecksumBypass bool `json:"-"`
```
Comment (gloriousCode) [inline exchanges/orderbook/orderbook_types.go:90]: I would prefer this to be `HasChecksumValidation`. This name implies that you are bypassing a checksum which is not what this is doing. I was going to suggest `ValidationBypass` but `HasChecksumValidation` is more specific to why you would ever skip validating the orderbook

**Review by gloriousCode — CHANGES_REQUESTED** [review]
Looks good!

File: exchanges/orderbook/orderbook_types.go (hunk 1 above)
Comment (shazbert) [reply exchanges/orderbook/orderbook_types.go]: Changed

Comment (gloriousCode) [reply exchanges/orderbook/orderbook_types.go]: Superfast, thanks!

**Review by thrasher- — APPROVED** [review]
utACK but good looks good!

**Review by gloriousCode — APPROVED** [review]
tACK! Nice work speeding it up further
# 🐎 💨 💨 💨 💨 💨 💨 💨 💨 💨 💨 💨 💨 
#         🏎️ 💨 💨 💨 💨 💨 💨 💨 💨 💨 💨 💨

**Review by MadCozBadd — APPROVED** [review]
tACK!!!!

**Pull request merged**